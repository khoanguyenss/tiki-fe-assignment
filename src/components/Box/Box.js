import React from 'react';
import PropTypes from 'prop-types';
import { BoxStyled } from './Box.styled';

const renderCellData = (data) => {
  if (!data) return '';
  if (data < 0) return 'x';
  return data;
};

const Box = ({ box, openBox }) => {
  if (box.isOpen) {
    return (
      <BoxStyled>
        {renderCellData(box.data)}
      </BoxStyled>
    );
  }
  return (
    <BoxStyled
      isClose
      onClick={
        () => openBox(box)
      }
    />
  );
};

Box.propTypes = {
  box: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number,
    data: PropTypes.number,
    isOpen: PropTypes.bool,
  }).isRequired,
  openBox: PropTypes.func.isRequired,
};

export default Box;
