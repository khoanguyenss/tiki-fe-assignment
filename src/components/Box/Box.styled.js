import styled from 'styled-components';

export const BoxStyled = styled.div`
  width: 35px;
  height: 35px;
  border: 1px solid #000;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
`;
