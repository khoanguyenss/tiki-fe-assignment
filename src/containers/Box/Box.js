import React from 'react';
import Box from '../../components/Box';
import {
  actions as MineSweeperActions,
} from '../../reducers/minesweeper';
import { useActions } from '../../utils/hooks';

const BoxContainer = (props) => {
  const openBox = useActions(
    MineSweeperActions.openBox,
  );

  return (
    <Box
      {...props}
      openBox={openBox}
    />
  );
};

export default BoxContainer;
