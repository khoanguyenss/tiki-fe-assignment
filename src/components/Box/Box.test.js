import React from 'react';
import { mount } from 'enzyme';
import Box from './Box';
import { BoxStyled } from './Box.styled';

describe('<Box />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const openBoxMock = jest.fn();

  const Component = mount(
    <Box
      openBox={openBoxMock}
      box={{
        x: 1,
        y: 1,
        data: 0,
        isOpen: false,
      }}
    />
    ,
  );

  it('should render', () => {
    expect(Component.length).toBe(1);
  });
  it('match snapshot', () => {
    expect(Component).toMatchSnapshot();
  });
  it('should call openBox when clicked', () => {
    const BoxStyledComp = Component.find(BoxStyled).at(0);
    BoxStyledComp.simulate('click');

    expect(openBoxMock).toBeCalledTimes(1);
  });
});
