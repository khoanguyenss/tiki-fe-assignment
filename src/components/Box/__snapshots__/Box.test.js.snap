// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> match snapshot 1`] = `
<Box
  box={
    Object {
      "data": 0,
      "isOpen": false,
      "x": 1,
      "y": 1,
    }
  }
  openBox={[MockFunction]}
>
  <styled.div
    isClose={true}
    isMine={0}
    onClick={[Function]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "kfvIne",
            "rules": Array [
              "
  width: 35px;
  height: 35px;
  border: 1px solid #000;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  
  ",
              [Function],
              "
  
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isClose={true}
      isMine={0}
      onClick={[Function]}
    >
      <div
        className="sc-bdVaJa kfvIne"
        onClick={[Function]}
      />
    </StyledComponent>
  </styled.div>
</Box>
`;
